guides/cheatsheets/git.md

```diff
@@ -0,0 +1,283 @@
+```
+git checkout -b [branch_name]
+```
+
+*Merge the specified branch’s history into the current one*.
+```
+git merge [branch]
+```
+
+*Show all commits in the current branch’s history*.
+```
+git log
+```
+----------
+## **INSPECT & COMPARE**
+----------
+Examining logs, diffs and object information.
+
+*Show the commit history for the currently active branch*.
+```
+git log 
+```
+
+*Show the commits on branchA that are not on branchB*.
+```
+git log branchB..branchA
+```
+
+*Show the commits that changed file, even across renames*.
+```
+git log --follow [file]
+```
+
+*Get the author of a commit*.
+```
+git blame [commit_hash]
+```
+
+*Show the diff of what is in branchA that is not in branchB*.
+```
+git diff branchB...branchA
+```
+
+*Show any object in Git in human-readable format*.
+```
+git show [SHA]
+```
+----------
+## **TRACKING PATH CHANGES**
+----------
+Versioning file removes and path changes.
+
+*Delete the file from project and stage the removal for commit*.
+```
+git rm [file]
+```
+
+*Change an existing file path and stage the move*.
+```
+git mv [existing-path] [new-path]
+```
+
+*Show all commit logs with indication of any paths that moved*.
+```
+git log --stat -M
+```
+
+----------
+## **IGNORING PATTERNS**
+----------
+Preventing unintentional staging or commiting of files.
+
+
+*Save a file with desired paterns as .gitignore with either direct string
+matches or wildcard globs*.
+```
+logs/
+*.notes
+pattern*/
+```
+
+*System wide ignore patern for all local repositories*.
+```
+git config --global core.excludesfile [file]
+```
+
+----------
+## **SHARE & UPDATE**
+----------
+Retrieving updates from another repository and updating local repos.
+
+
+*Add a git URL as an alias*.
+```
+git remote add [alias] [url]
+```
+
+*Fetch down all the branches from that Git remote*.
+```
+git fetch [alias]
+```
+
+*Merge a remote branch into your current branch to bring it up to date*.
+```
+git merge [alias]/[branch]
+```
+*Transmit local branch commits to the remote repository branch*.
+```
+git push [alias] [branch]
+```
+
+*Fetch and merge any commits from the tracking remote branch*.
+```
+git pull
+```
+----------
+## **REWRITE HISTORY**
+----------
+Rewriting branches, updating commits and clearing history.
+
+*Apply any commits of current branch ahead of specified one*.
+```
+git rebase [branch]
+```
+*Clear staging area, rewrite working tree from specified commit*.
+```
+git reset --hard [commit]      !--- USE WITH CAUTION ---!
+```
+
+----------
+## **TEMPORARY COMMITS**
+----------
+Temporarily store modified, tracked files in order to change branches.
+
+*Save modified and staged changes*.
+```
+git stash
+```
+
+*List stack-order of stashed file changes*.
+```
+git stash list
+```
+
+*Write working from top of stash stack*.
+```
+git stash pop
+```
+
+*Discard the changes from top of stash stack*.
+```
+git stash drop
+```
```

Comment:
Should we link to the "full" git guide from this page? [Git Technical Guide](https://github.com/TheDataShed/EngineeringHandbook/blob/main/guides/technical-guides/git.md)